application/src/main/java/org/opentripplanner/model/PathTransfer.java

```diff
@@ -46,6 +56,14 @@ public List<Edge> getEdges() {
     return this.edges;
   }
 
+  public EnumSet<StreetMode> getModes() {
+    return this.modes;
+  }
+
+  public boolean addMode(StreetMode mode) {
```

Comment:
I don't think we want to make this class mutable. Can you create a copy method, for example `withAddedMode` that returns a new instance instead?